File: src/main/java/ru/mrak/service/dto/BookCreateDTO.java
package ru.mrak.service.dto;

import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.NotNull;

/**
 * Объект для создания новой книги
 */
@Getter
@Setter
public class BookCreateDTO {

    @NotNull
    private String title;

    private String author;

    @NotNull
    private String text;

    private String pictureId;
}

File: src/main/webapp/src/model/bookCreateDto.ts
export interface BookCreate {
  title?: string;
  author?: string;
  text?: string | null | ArrayBuffer;
  pictureId?: string;
  pictureUrl?: string;
}

export class BookCreateDto implements BookCreate {
  constructor (
    public title?: string,
    public author?: string,
    public text?: string | null | ArrayBuffer,
    public pictureId?: string,
    public pictureUrl?: string
  ) {
    this.pictureId = this.pictureId || ''
    this.pictureUrl = this.pictureUrl || ''
  }
}
